framework/src/loops/ComputeInitialConditionThread.C

```diff
@@ -93,8 +93,23 @@ ComputeInitialConditionThread::operator()(const ConstElemRange & range)
             e, "Cyclic dependency detected in object ordering");
       }
 
-      for (auto ic : vec)
-        ic->compute();
+      /* Logic for running only old state initial conditions if they exist, but
+          running ICs normally if the old state ICs don't exist.
+      */
+      bool break_loop = false;
+      int loop_state;
+      for (loop_state = _fe_problem._global_current_state; loop_state >= 0; loop_state--)
+      {
+        for (auto ic : vec)
+          if (ic->_my_state == loop_state)
+          {
+            ic->compute();
+            break_loop = true;
+          }
+
+        if (break_loop == true)
+          break;
+      }
```

Comment:
You could help me understand this better... Why do we immediately break after one IC is computed? What if I have multiple IC objects for the same variable with the same state, just defined on different blocks/boundaries, does this mean only one of them gets computed?

As a side note, directly accessing members with an underscore prefix is not the best practice in MOOSE.

Reply:
We are not breaking after one IC is computed. Lines 104-108 compute for every ic in the vector before it has a chance to break. 

However, I am open to suggestions for better logic. Just remember that vec is a vector of ic's for only ONE variable. So other variables will not be affected.

Reply:
I used a member variable _fe_problem._global_current_state because we needed to transfer the loop state from FEProblemBase::projectSolution(). Is there another location for a loop variable that would be available in both files? That was why I chose to implement it this way. But I am open to suggestions.

Reply:
I need to look at this more closely. I suggest other reviewers should also double check this.

Reply:
I would love to implement it a cleaner way. Thanks for checking it out.